sdk/nodejs/automation/localWorkspace.ts

```diff
@@ -815,9 +832,17 @@ export class LocalWorkspace implements Workspace {
         return;
     }
     private async getPulumiVersion(minVersion: semver.SemVer) {
-        const result = await this.runPulumiCmd(["version"]);
         const optOut = !!this.envVars[SKIP_VERSION_CHECK_VAR] || !!process.env[SKIP_VERSION_CHECK_VAR];
-        const version = parseAndValidatePulumiVersion(minVersion, result.stdout.trim(), optOut);
+        let version: semver.SemVer | null;
+        if (this.pulumi) {
+            version = this.pulumi.version;
+        } else {
+            const result = await this.runPulumiCmd(["version"]);
+            version = semver.parse(result.stdout.trim());
+        }
+        if (!optOut) {
+            validatePulumiVersion(minVersion, version, optOut);
+        }
```

Comment:
I think we can either get rid of this `if` here, or of the `optOut` parameter in `validatePulumiVersion`.